src/hardware/voodoo.cpp

```diff
@@ -6996,6 +7054,54 @@ static constexpr uint32_t voodoo_r(const uint32_t addr)
 	return 0xffffffff;
 }
 
+// Get the number of additional threads to use for Voodoo work based on the
+// user's conf setting. By default we use up to 7 threads however the user can
+// customize this.
+
+static int get_num_threads()
+{
+	auto get_auto_threads = []() {
+		// Sweet spot found to be seven threads based on benchmarks:
+		// https://github.com/schellingb/dosbox-pure/issues/300#issuecomment-2144050898
+		constexpr auto MinThreads = 0;
+		constexpr auto MaxThreads = 7;
+		return std::clamp(SDL_GetCPUCount() - 1, MinThreads, MaxThreads);
+	};
+
+	auto parse_positive_base10_int = [](const std::string& str) {
+		constexpr auto Base10 = 10;
+		const auto maybe_int  = parse_int(str, Base10);
```

Comment:
This is noisy, just do `parse_int`; base10 is the default (`const int base = 10`).

Reply:
Thanks, added.